packages/calcite-components/src/components/filter/filter.tsx

```diff
@@ -219,16 +235,16 @@ export class Filter
 
   clear = (): void => {
     this.value = "";
-    this.filter("", true);
+    this.filterDebounced("", true);
     this.setFocus();
   };
 
-  updateFiltered(filtered: any[], emit = false): void {
-    this.filteredItems.length = 0;
-    this.filteredItems = this.filteredItems.concat(filtered);
+  updateFiltered(filtered: object[], emit = false, callback?: () => void): void {
```

Comment:
WDYT about `onFilter` or something along those lines for the callback name? Up to you.